corehq/apps/users/views/mobile/groups.py

```diff
@@ -20,6 +32,79 @@ def _get_sorted_groups(domain):
     )
 
 
+def get_group_or_404(domain, group_id):
+    try:
+        group = Group.get(group_id)
+        assert group.doc_type.startswith('Group')
+        assert group.domain == domain
```

Comment:
You should use an assert to verify conditions that you expect to be true, and no known end-user action could (should) ever be able to impact the outcome of the assertion. An assertion failure should never be depended on for program flow.